locator/network_topology_strategy.cc

```diff
@@ -554,6 +556,41 @@ tablet_replica_set network_topology_strategy::drop_tablets_in_dc(schema_ptr s, c
     return filtered;
 }
 
+sstring network_topology_strategy::sanity_check_read_replicas(const effective_replication_map& erm,
+                                                              const inet_address_vector_replica_set& read_replicas) const {
+    const auto& topology = erm.get_topology();
+
+    struct rf_node_count {
+        size_t replication_factor{0};
+        size_t node_count{0};
+    };
+
+    absl::flat_hash_map<sstring, rf_node_count> data_centers_replication_factor;
+    std::ranges::for_each(read_replicas, [&data_centers_replication_factor, &topology, this](const auto& node) {
+        auto res = data_centers_replication_factor.emplace(topology.get_datacenter(node), rf_node_count{0, 0});
+        if (res.second) {
+            // For new item add replication factor.
+            res.first->second.replication_factor = get_replication_factor(res.first->first);
+        }
+        ++res.first->second.node_count;
+    });
+
+    sstring error_msg;
+
+    std::ranges::any_of(data_centers_replication_factor, [&error_msg](const auto& item) {
```

Comment:
the bool result of the `any_of()` is not used, so it's usage looks a bit strange here - there might be a better named algo to use (maybe `find_if`?) or even something that would return the message directly (transform / filter)?

`std:P:ranges::find_if` looks quite fitting for this task, if we are looking for the first bad count (and then checking the returned iterator and setting the error if not end, and not having to capture the error_msg)

Reply:
Hmm, but in your case would be 2 branches instead of one, and besides `any_of` does exactly what was intended - it checks all the values and stops on the first one. Since I have the value in the string, I do not need to check anything after it finished, and return the result string right away.

Reply:
maybe even a range for loop might work better here?
(the advantage being that you can return the error message directly via return from the range loop instead of having to store and capture it)

Reply:
Changed to a range loop.